src/main/java/payapal/Perfection.java

```diff
@@ -0,0 +1,42 @@
+package main.java.payapal;
+
+public class Perfection {
+
+    private static Perfection perf;
```

Comment:
can you please clarify why are you using static here?

In case of Singleton, can you please, consider avoid the null check using a private construct for the Singleton and also providing other approach considering multiple Threads?